src/main/java/org/humancellatlas/ingest/migrations/MongoChangeLog.java

```diff
@@ -183,4 +189,16 @@ public void renameSubmissionStates(MongoDatabase db) {
         update.add(new Document("$set", Document.parse("{ 'submissionState': 'METADATA_INVALID' }")));
         db.getCollection("submissionEnvelope").updateMany(filter, update);
     }
+
+    @ChangeSet(order = "2022-05-06",
+               id = "add missing dataFileUuid for File documents with a unique uuid. dcp-764",
+               author = "amnon@ebi.ac.uk")
+    public void addMissingDataFileUuidToFiles(MongoDatabase db) {
+        MongoCollection<Document> files = db.getCollection("file");
+        files.find(eq("dataFileUuid", null))
+             .forEach((Consumer<? super Document>) (Document file) ->
+                files
+                    .updateOne(eq("_id", file.get("_id")),
+                               Updates.set("dataFileUuid", UUID.randomUUID())));
```

Comment:
I think you can avoid a forEach here and have mongo do the work for you:
``` mongo.java
        Document filter = Document.parse("{dataFileUuid: {$exists: 0}}");
        List<Document> update = new ArrayList<>();
        update.add(new Document("$set", Document.parse("{ dataFileUuid: UUID()}")));
        db.getCollection("file").updateMany(filter, update);
```
I'm going to test it.

Reply:
Would the `UUID()` in your suggestion run for every document or just once?
If your suggestion indeed works it would be theoretically more correct and faster, however, since this migration is a one time thing that would update only ~16k records, it does not really matter which syntax we choose.

Reply:
We have had problems with mongo migrations taking so long that kubernettes kills to pod for not starting quickly enough.

Reply:
This is fine, only takes 20 seconds to update all 16 thousand records.